### PR TITLE
Feature: caching improvements to ci

### PR DESCRIPTION
**Pull request by lipkau**
### Description

Small improvements to CI pipeline:
* added PS module caching
* added upload of test results in ci
* added yaml file settings

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] My code follows the code style of this project.
- [ ] I have added Pester Tests that describe what my changes should do.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by johnheusinger — APPROVED** [review]
There's an issue with the CODEOWNERS file - has the `ci-managers` group been given access to the repo? LGTM aside from that.

Thanks for the invite!

**Comment by lipkau** [comment]
> There's an issue with the CODEOWNERS file 

what is the issue you found?

> has the `ci-managers` group been given access to the repo?

I did it today, yes

**Comment by johnheusinger** [comment]
> > There's an issue with the CODEOWNERS file
> 
> what is the issue you found?
> 
> > has the `ci-managers` group been given access to the repo?
> 
> I did it today, yes

When I look at the file [here](https://github.com/AtlassianPS/JiraPS/pull/481/files), I see the following:

```
Unknown owner on line 5: make sure the team @atlassianps/ci-managers exists, is publicly visible, and has write access to the repository
…ithub/workflows               @atlassianps/ci-managers
Unknown owner on line 6: make sure the team @atlassianps/ci-managers exists, is publicly visible, and has write access to the repository
…ols/                          @atlassianps/ci-managers
Unknown owner on line 7: make sure the team @atlassianps/ci-managers exists, is publicly visible, and has write access to the repository
…raPS.build.ps1                @atlassianps/ci-managers
Unknown owner on line 8: make sure the team @atlassianps/ci-managers exists, is publicly visible, and has write access to the repository
…SScriptAnalyzerSettings.psd1  @atlassianps/ci-managers
```

File: .github/workflows/pr_review.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  label_issue:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Add default reviewers to PR"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ISSUE_URL: ${{ github.event.pull_request.html_url }}
+        run: |
+          gh pr edit $PR --add-reviewer @atlassianps/reviewers
```
Comment (johnheusinger) [inline .github/workflows/pr_review.yml:16]:
```suggestion
          ISSUE_URL: ${{ github.event.pull_request.html_url }}
        run: |
          gh pr edit $ISSUE_URL --add-reviewer @atlassianps/reviewers
```

I think you want the variable names on lines 14 and 16 to match

**Pull request merged**